### PR TITLE
[WIP] feat(product): add/edit models for bundled products

### PR DESCRIPTION
**Pull request by lderrickable**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## What is the new behavior?
Added models and factories for bundled product. Edited DaffProduct model to work with bundled products.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by lderrickable** [comment]
I just made this PR the model and factory changes.

**Comment by lderrickable** [comment]
build isn't failing because of these changes

**Comment by lderrickable** [comment]
This PR addresses the failed build and other issues: https://github.com/graycoreio/daffodil/pull/682

File: libs/product/src/models/bundled-product.ts [hunk 1]
```diff
@@ -1,18 +1,12 @@
 import { DaffProduct } from './product';
 import { DaffBundledProductItem } from './bundled-product-item';
 
-export enum MagentoPriceViewEnum {
```
Comment (damienwebdev) [inline libs/product/src/models/bundled-product.ts]: Remove these types, we don't have a purpose for them in Daffodil yet.

File: libs/product/src/models/bundled-product.ts [hunk 2]
```diff
@@ -1,18 +1,12 @@
 import { DaffProduct } from './product';
```
Comment (damienwebdev) [inline libs/product/src/models/bundled-product.ts]: file rename to composite.

File: libs/product/src/models/bundled-product-item.ts [hunk 3]
```diff
@@ -6,18 +6,23 @@ export enum DaffPriceTypeEnum {
 	dynamic = 'DYNAMIC'
```
Comment (damienwebdev) [inline libs/product/src/models/bundled-product-item.ts]: rename file to composite

File: libs/product/src/models/bundled-product-item.ts [hunk 4]
```diff
@@ -6,18 +6,23 @@ export enum DaffPriceTypeEnum {
...
 }
 
+export enum DaffBundledProductItemType {
```
Comment (damienwebdev) [inline libs/product/src/models/bundled-product-item.ts]: This is an enum not a type, the name should reflect this. It's also a `Composite` product.

File: libs/product/src/models/bundled-product-item.ts [hunk 5]
```diff
@@ -6,18 +6,23 @@ export enum DaffPriceTypeEnum {
...
+	radio = 'RADIO',
+	checkbox = 'CHECKBOX',
+	select = 'SELECT'
+}
+
 export interface DaffBundledProductItem {
```
Comment (damienwebdev) [inline libs/product/src/models/bundled-product-item.ts]: `Composite`

File: libs/product/src/models/bundled-product.ts [hunk 6]
```diff
@@ -1,18 +1,12 @@
...
 
-export enum MagentoBundleItemsEnum {
-	together = 'TOGETHER',
-	separately = 'SEPARATELY'
-}
-
 export interface DaffBundledProduct extends DaffProduct {
-	price_view: MagentoPriceViewEnum;
-	ship_bundle_items: MagentoBundleItemsEnum;
+	price_view: DaffBundledProductPriceViewEnum;
```
Comment (damienwebdev) [inline libs/product/src/models/bundled-product.ts]: We don't need this key at the moment.

File: libs/product/testing/src/factories/bundled-product.factory.ts [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+import { MockProduct } from './product.factory';
+
+/**
+ * Mocked DaffBundledProduct object.
+ */
+export class MockBundledProduct extends MockProduct implements DaffBundledProduct {
+	__typename = DaffProductTypeEnum.Composite;
+	price_view = DaffBundledProductPriceViewEnum.as_low_as;
+	items = [{
+		bundle_id: faker.random.alphaNumeric(10),
```
Comment (damienwebdev) [inline libs/product/testing/src/factories/bundled-product.factory.ts]: We don't need this key.

File: libs/product/src/models/bundled-product-item.ts [hunk 8]
```diff
@@ -6,18 +6,23 @@ export enum DaffPriceTypeEnum {
...
 	bundle_id: string;
```
Comment (damienwebdev) [inline libs/product/src/models/bundled-product-item.ts]: we don't need this

File: libs/product/src/models/bundled-product-item.ts [hunk 9]
```diff
@@ -6,18 +6,23 @@ export enum DaffPriceTypeEnum {
...
 	bundle_option_id: number;
```
Comment (damienwebdev) [inline libs/product/src/models/bundled-product-item.ts]: this should just be `id`

File: libs/product/src/models/bundled-product.ts (hunk 1 above)
Comment (lderrickable) [reply libs/product/src/models/bundled-product.ts]: done

File: libs/product/src/models/bundled-product.ts (hunk 2 above)
Comment (lderrickable) [reply libs/product/src/models/bundled-product.ts]: done

File: libs/product/src/models/bundled-product-item.ts (hunk 3 above)
Comment (lderrickable) [reply libs/product/src/models/bundled-product-item.ts]: done

File: libs/product/src/models/bundled-product-item.ts (hunk 4 above)
Comment (lderrickable) [reply libs/product/src/models/bundled-product-item.ts]: done

File: libs/product/src/models/bundled-product-item.ts (hunk 5 above)
Comment (lderrickable) [reply libs/product/src/models/bundled-product-item.ts]: done

File: libs/product/src/models/bundled-product.ts (hunk 6 above)
Comment (lderrickable) [reply libs/product/src/models/bundled-product.ts]: removed

File: libs/product/testing/src/factories/bundled-product.factory.ts (hunk 7 above)
Comment (lderrickable) [reply libs/product/testing/src/factories/bundled-product.factory.ts]: removed

File: libs/product/src/models/bundled-product-item.ts (hunk 8 above)
Comment (lderrickable) [reply libs/product/src/models/bundled-product-item.ts]: removed

File: libs/product/src/models/bundled-product-item.ts (hunk 9 above)
Comment (lderrickable) [reply libs/product/src/models/bundled-product-item.ts]: changed

**Comment by lderrickable** [comment]
I had accidentally committed some files I didn't need by accident. I force pushed to remove them

**Comment by damienwebdev** [comment]
@lderrickable can you split the "type" changes out of this into another PR?

**Pull request closed without merging**